### PR TITLE
EOSPAC vector performance

### PR DESCRIPTION
**Pull request by rbberger**
## PR Summary

- add check that EOSPAC vector functions are actually faster (profile_eospac)
- prevents slower template to become active in some cases
- adds a warning if slower non scratch version gets called
- Closes https://github.com/lanl/singularity-eos/issues/257

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
Thanks for adding this fix, @rbberger ! Can you add the PR to the changelog as well? After @jhp-lanl approves, I think this is ready for merge.

**Review by jhp-lanl — APPROVED** [review]
Looks good to me. Once you add a changelog entry I think this is good to go. Thanks @rbberger !

**Comment by rbberger** [comment]
@Yurlungur @jhp-lanl done.

**Comment by Yurlungur** [comment]
Waiting for tests to pass, then will merge.

**Pull request merged**
